cmd/cmdtest/testutils.go

```diff
@@ -34,6 +35,18 @@ type Repository struct {
 	URL  string
 }
 
+type RepositoryFile struct {
+	APIVersion   string
+	Generated    string
+	Repositories []Repository
+}
+
+type Stack struct {
+	ID          string
+	Version     string
+	Description string
```

Comment:
Yes, I have redesigned the structure and will upload my changes. 

Regarding the structs, why would it be beneficial to maintain a different struct in the test? Don't we want to test that the structure of these structs is correctly outputted? If we change the structure of a struct, then our test may not be testing the whole contents of the struct, but only the fields specified in the test struct?

Reply:
It depends on what you are trying to test. Generally you shouldn't reuse the code you are trying to test in the test itself. Ideally we would even use a different (or custom) json/yaml parser library to increase the chances of catching a bug in the parser. 

I could argue that if you change the structure of the command output, the tests should break. This makes the developer aware that they are introducing a potentially breaking change and they have to consciously update the test to confirm they want to do that.

Practically speaking, I'm okay with reusing some structs in our tests, especially since we are more focused on testing the command arguments and not necessarily parser library.